master/internal/api_workspace.go

```diff
@@ -100,6 +129,16 @@ func (a *apiServer) GetWorkspaceProjects(ctx context.Context,
 		return nil, err
 	}
 
+	user, err := a.CurrentUser(ctx, &apiv1.CurrentUserRequest{})
+	if err != nil {
+		return nil, err
+	}
+	resp.Projects, err = workspace.AuthZProvider.Get().
```

Comment:
as currently implemented, `FilterWorkspaceProjects` is letting everything through, no?  We should have a step before this which checks whether the user is allowed to read the Workspace.

Reply:
Yep that is a good point added a check since it doesn't make sense to view workspace projects if you can't view the workspace.